test/bidir-full-sync-tests.js

```diff
@@ -0,0 +1,45 @@
+var events = require('./events');
+var test2 = require('./test-util').test2;
+var dsl = require('./ringpop-assert');
+var prepareCluster = require('./test-util').prepareCluster;
+var prepareWithStatus = require('./test-util').prepareWithStatus;
+var getClusterSizes = require('./it-tests').getClusterSizes;
+var _ = require('lodash');
+
+test2('bidirectional full sync test', getClusterSizes(5), 20000,
+    prepareCluster(function(t, tc, n) {
+	    return [
+            dsl.assertStats(t, tc, n+1, 0, 0),
+            dsl.waitForEmptyPing(t, tc),
+
+            // create faulties so that the membership of the faulty nodes is
+            // out of sync with the membership of the SUT
+            createPartition(t, tc),
+
+            // this causes the SUT to full sync since the chechsums do not match
+            dsl.sendPing(t, tc, 0),
+            dsl.waitForPingResponse(t, tc, 0),
+
+            // the SUT sends a join request to perform a bidirectional full sync
+            dsl.waitForJoins(t, tc, 1),
+
+            // wait for the SUT to receive and process the join response
+            dsl.wait(100),
```

Comment:
This assumes that in 100ms it will process the join request, which might be false in CI.

Can we instead poll for stats every X (say 10ms), and, after Y times we didn't receive the right answer, fail then? This will speed up the test in the "fast machine" case (when it takes 2ms to process the join request), and will still work in travis.

In that case, we can extend the timeout even more -- to, say, 5 seconds, since on the "good" case it will still take much less.